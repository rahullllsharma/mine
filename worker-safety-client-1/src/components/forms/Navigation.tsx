import cx from "classnames";

export type Props = {
  wizardNavigation: React.ReactElement;
  children: React.ReactNode;
};

export function View(props: Props): JSX.Element {
  return (
    <div
      className={cx(
        "flex flex-col md:flex-row md:gap-4 flex-1",
        "w-full max-w-screen-lg",
        "overflow-hidden"
      )}
    >
      <nav
        className={cx(
          "flex flex-col items-start gap-2",
          "w-full md:min-w-[254px] md:max-w-[254px]",
          "px-3 md:py-6",
          "overflow-y-auto",
          "md:bg-white",
          "h-16 md:h-auto"
        )}
      >
        {props.wizardNavigation}
      </nav>
      <section
        className={cx(
          "flex-1 w-full",
          "overflow-y-auto",
          "bg-white",
          "md:py-8 md:px-6"
        )}
      >
        {props.children}
      </section>
    </div>
  );
}
