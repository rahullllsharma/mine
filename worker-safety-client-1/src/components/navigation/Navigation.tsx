import type {
  RenderOptionFn,
  SelectOption,
} from "@/components/shared/select/Select";
import type { NavItemProps } from "./navItem/NavItem";
import cx from "classnames";
import { Select } from "@/components/shared/select/Select";
import { defaultRenderOptionFn } from "../shared/select/selectPrimary/SelectPrimary";
import Sidebar from "../sidebar/Sidebar";
import NavItem from "./navItem/NavItem";

export type NavigationStatus =
  | "default"
  | "completed"
  | "saved"
  | "saved_current"
  | "error";

export type SelectStatusAndIcon =
  | {
      status: "default";
      icon: "circle";
    }
  | {
      status: "completed";
      icon: "circle_check";
    }
  | {
      status: "error";
      icon: "error";
    };

export type NavigationOption = Pick<
  NavItemProps,
  "name" | "icon" | "iconSize" | "status" | "isSubStep"
> & {
  id: number;
  onSelect?: () => void;
};

export type NavigationProps = {
  options: NavigationOption[];
  selectedIndex: number;
  withStatus?: boolean;
  sidebarClassNames?: string;
  selectClassNames?: string;
  onChange: (index: number) => void;
};

const renderOptionFn: RenderOptionFn<NavigationOption> = ({
  listboxOptionProps: { selected },
  option: { name, icon, status },
}) => (
  <NavItem
    as="li"
    icon={icon}
    name={name}
    status={status}
    markerType="left"
    isSelected={selected}
  />
);

export default function Navigation({
  options,
  selectedIndex,
  withStatus = false,
  sidebarClassNames,
  selectClassNames,
  onChange,
}: NavigationProps): JSX.Element {
  const navigationSelectHandler = (option: SelectOption<NavigationOption>) => {
    const index = options.findIndex(opt => opt.id === option.id);
    onChange(index);
  };

  return (
    <>
      <div className={cx("hidden md:block", sidebarClassNames)}>
        <Sidebar
          options={options}
          selectedIndex={selectedIndex}
          onChange={onChange}
        />
      </div>
      <Select
        className={cx("md:hidden block", selectClassNames)}
        options={options}
        onSelect={navigationSelectHandler}
        defaultOption={options[selectedIndex]}
        optionsClassNames="py-2"
        renderOptionFn={withStatus ? renderOptionFn : defaultRenderOptionFn}
      />
    </>
  );
}
