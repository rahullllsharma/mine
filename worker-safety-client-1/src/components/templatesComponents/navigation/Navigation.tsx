import type { ChangeEvent } from "react";
import router from "next/router";
import ButtonPrimary from "../../shared/button/primary/ButtonPrimary";
import ButtonSecondary from "../../shared/button/secondary/ButtonSecondary";
import Input from "../../shared/input/Input";

type ListNavigationParams = {
  toggleFilters: () => void;
  totalFilterSelected?: number;
  searchValue: string;
  onSearchChange?: (event: ChangeEvent<HTMLInputElement>) => void;
};

const ListNavigationBar = ({
  toggleFilters,
  totalFilterSelected,
  searchValue,
  onSearchChange,
}: ListNavigationParams): JSX.Element => {
  const createTemplate = () => {
    return router.push("templates/create");
  };

  return (
    <section className="flex mb-6">
      <div className="flex flex-1">
        <h4 className="text-neutral-shade-100">Templates</h4>
        <Input
          containerClassName="hidden lg:flex lg:w-60 ml-6"
          placeholder={`Search Templates`}
          value={searchValue}
          onChange={onSearchChange}
          icon="search"
        />
        <ButtonSecondary
          iconStart="filter"
          label={
            totalFilterSelected ? `Filters (${totalFilterSelected})` : "Filters"
          }
          controlStateClass="text-base p-1.5"
          className="text-neutral-shade-100 flex-shrink-0 ml-6 h-[2.3rem]"
          onClick={toggleFilters}
        />
      </div>
      <div className="flex">
        <ButtonPrimary label="Create Template" onClick={createTemplate} />
      </div>
    </section>
  );
};

export { ListNavigationBar };
