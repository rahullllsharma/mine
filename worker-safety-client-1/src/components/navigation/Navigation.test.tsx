import type { NavigationOption } from "./Navigation";
import { render, screen, fireEvent } from "@testing-library/react";
import Navigation from "./Navigation";

const options: NavigationOption[] = [
  {
    id: 0,
    name: "Details",
    icon: "settings_filled",
  },
  {
    id: 1,
    name: "Locations",
    icon: "location",
  },
  {
    id: 2,
    name: "History",
    icon: "history",
  },
];

const mockOnChange = jest.fn();

describe(Navigation.name, () => {
  describe("when it renders", () => {
    it("should match the snapshot", () => {
      const { asFragment } = render(
        <Navigation
          options={options}
          onChange={mockOnChange}
          selectedIndex={0}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when the "onChange" callback is called from the select component', () => {
    it("should return the index of the option", () => {
      render(
        <Navigation
          options={options}
          onChange={mockOnChange}
          selectedIndex={0}
        />
      );
      fireEvent.click(screen.getByRole("button")); //the button is the trigger for the select
      fireEvent.click(screen.getByRole("option", { name: options[1].name }));
      expect(mockOnChange).toBeCalledWith(1);
    });
  });
});
