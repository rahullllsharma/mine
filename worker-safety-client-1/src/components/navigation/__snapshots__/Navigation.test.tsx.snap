// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation when it renders should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="hidden md:block"
  >
    <div
      aria-orientation="vertical"
      class="flex flex-col gap-2"
      role="tablist"
    >
      <button
        aria-selected="true"
        class="bg-neutral-shade-3"
        id="headlessui-tabs-tab-1"
        role="tab"
        tabindex="0"
        type="button"
      >
        <div
          class="relative flex items-center px-2.5 py-3 rounded h-12 w-full hover:shadow-10 border-2 text-neutral-shade-100 cursor-pointer border-neutral-shade-100"
        >
          <i
            aria-hidden="true"
            class="ci-settings_filled text-xl mr-1 text-neutral-shade-75"
          />
          <span
            class="text-base mx-1 truncate"
          >
            Details
          </span>
        </div>
      </button>
      <button
        aria-selected="false"
        class="bg-neutral-shade-3"
        id="headlessui-tabs-tab-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <div
          class="relative flex items-center px-2.5 py-3 rounded h-12 w-full hover:shadow-10 border-2 text-neutral-shade-100 cursor-pointer border-transparent"
        >
          <i
            aria-hidden="true"
            class="ci-location text-xl mr-1 text-neutral-shade-75"
          />
          <span
            class="text-base mx-1 truncate"
          >
            Locations
          </span>
        </div>
      </button>
      <button
        aria-selected="false"
        class="bg-neutral-shade-3"
        id="headlessui-tabs-tab-3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <div
          class="relative flex items-center px-2.5 py-3 rounded h-12 w-full hover:shadow-10 border-2 text-neutral-shade-100 cursor-pointer border-transparent"
        >
          <i
            aria-hidden="true"
            class="text-xl mr-1 text-neutral-shade-75"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12 3.75c-2.228 0-4.25.884-5.734 2.319l-.97-.969a.707.707 0 00-1.204.436L3.77 9.092a.707.707 0 00.768.768l3.556-.323a.707.707 0 00.436-1.205l-.94-.94A6.379 6.379 0 115.621 12a.936.936 0 10-1.871 0A8.25 8.25 0 1012 3.75zm0 3.743a.936.936 0 00-.936.936V13c0 .323.166.623.44.793l2.743 1.715a.936.936 0 00.992-1.587l-2.303-1.44V8.43A.936.936 0 0012 7.493z"
                fill-rule="evenodd"
              />
            </svg>
          </i>
          <span
            class="text-base mx-1 truncate"
          >
            History
          </span>
        </div>
      </button>
    </div>
  </div>
  <div
    class="relative text-neutral-shade-100 md:hidden block"
    data-testid="select"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="w-full px-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 border-neutral-shade-26"
      id="headlessui-listbox-button-4"
      type="button"
    >
      <span
        class="truncate"
      >
        Details
      </span>
      <i
        aria-hidden="true"
        class="ci-chevron_down text-neutral-shade-58 text-xl"
      />
    </button>
  </div>
</DocumentFragment>
`;
